### PR TITLE
Add Userflow to Colony

### LINKED ISSUE #2884: Add Userflow to Colony
### Story

As an onboarding team, we want to offer new users a guided onboarding experience so that we maximize our conversion of users onboarded.

As an onboarding team, we want to use guided onboarding experiences so that we reduce the churn of users who create coloneis.

### This issue is complete when...

- [ ] Userflow is setup for our colonies

We're planning on using it for when folks are interacting with https://xdai.colony.io/colony/* We've setup flows for setting up a colony, doing expenditures, installing motions & disputes via their no-code interface. We can disable / enable the flows from Userflow's control panel. 

Planning on using 2B installation: https://userflow.com/docs/dev/userflow-js-installation

Which is adding the following snippet:

```
<script>
  !function(){var e,t="undefined"==typeof window?{}:window,r=t.userflow;if(!r){var n="https://js.userflow.com/";r=t.userflow={_stubbed:!0};var o=t.USERFLOWJS_QUEUE=t.USERFLOWJS_QUEUE||[],s=function(e){r[e]=function(){var t=Array.prototype.slice.call(arguments);u(),o.push([e,null,t])}},a=function(e){r[e]=function(){var t,r=Array.prototype.slice.call(arguments);u();var n=new Promise((function(e,r){t={resolve:e,reject:r}}));return o.push([e,t,r]),n}},i=!1,u=function(){if(!i){i=!0;var e=document.createElement("script");e.async=!0;var r=t.USERFLOWJS_ENV_VARS||{};"es2020"===(r.USERFLOWJS_BROWSER_TARGET||function(e){for(var t=[[/Edg\//,/Edg\/(\d+)/,80],[/OPR\//,/OPR\/(\d+)/,67],[/Chrome\//,/Chrome\/(\d+)/,80],[/Safari\//,/Version\/(\d+)/,14],[/Firefox\//,/Firefox\/(\d+)/,74]],r=0;r<t.length;r++){var n=t[r],o=n[0],s=n[1],a=n[2];if(e.match(o)){var i=e.match(new RegExp(s));if(i&&parseInt(i[1],10)>=a)return"es2020";break}}return"legacy"}(navigator.userAgent))?(e.type="module",e.src=r.USERFLOWJS_ES2020_URL||n+"es2020/userflow.js"):e.src=r.USERFLOWJS_LEGACY_URL||n+"legacy/userflow.js",e.onerror=function(){i=!1,console.error("Could not load Userflow.js")},document.head.appendChild(e)}};s("_setTargetEnv"),s("init"),s("off"),s("on"),s("prepareAudio"),s("registerCustomInput"),s("remount"),s("reset"),s("setCustomInputSelector"),s("setCustomNavigate"),s("setCustomScrollIntoView"),s("setInferenceAttributeFilter"),s("setInferenceAttributeNames"),s("setInferenceClassNameFilter"),s("setScrollPadding"),s("setShadowDomEnabled"),s("setUrlFilter"),a("endAll"),a("endAllFlows"),a("endChecklist"),a("group"),a("identify"),a("identifyAnonymous"),a("start"),a("startFlow"),a("startWalk"),a("track"),a("updateGroup"),a("updateUser"),e=!1,r["isIdentified"]=function(){return e}}}();

  userflow.init('ct_snrheuld3nbe3exppc4foxcb54')
</script>
```

### PR DESCRIPTION
**Pull request by arrenv**
## Description

Adds the 3rd-Party Userflow.com package for no-code user onboarding in the Dapp.

**New stuff** ✨

* Adds userflow-js package to Dapp
* Adds new environment variable for linking to the account
* Initiates the Userflow resource within the Dapp

Resolves #2884

### CODE REVIEW

**Comment by arrenv** [comment]
This PR is on hold until GDPR and a few other things are figured out within the Dapp.

**Comment by rdig** [comment]
@arrenv can please find out the state of GDPR for this? _(I remember hearing something about that being solved, but not sure)_

So we can determine what to do with this PR

File: src/routes/AlwaysAccesibleRoute.tsx [hunk 1]
```diff
@@ -13,6 +14,11 @@ interface Props extends RouteProps {
   routeProps?: RouteComponentProps | routePropsFn;
 }
 
+// Initiate Userflow
+if (process.env.USERFLOW_TOKEN) {
+  userflow.init(process.env.USERFLOW_TOKEN);
+}
```
Comment (rdig) [inline src/routes/AlwaysAccesibleRoute.tsx:20]: Why put this in the route and not inside the app index? _(as that is the initial entry point)_

Comment (arrenv) [reply src/routes/AlwaysAccesibleRoute.tsx:20]: I originally had it in the root index.ts, however, thought it was matching the pattern of the app by putting it alongside FeedbackWidget and BetaCautionAlert components.

I will move it into the index.ts though, seeing as though you raised it.

**Comment by rdig** [comment]
> > @arrenv can please find out the state of GDPR for this? _(I remember hearing something about that being solved, but not sure)_
> > So we can determine what to do with this PR
> 
> Seeing as though we are not collecting any personal data, we should not have to include GDPR within the dapp.
> 
> > If you have any EU personal data in your Userflow account, such as names, email addresses, ID numbers, or… anything personally identifiable, then GDPR applies.
> 
> https://userflow.com/policies/privacy-regulations

Haha, sure, but I didn't meant **you** to solve this, just go ping people responsible for this, most likely @thycommodore and see where they stand on this, and if we can proceed with finishing up the implementation

**Comment by thycommodore** [comment]
**tl;dr — we're clear imo**

In an alternative universe, I might have been a lawyer... but as an amateur here's how I understand this.

We're not passing Userflow any user data. Userflow is not collecting any user data & passing it back to us. 

It appears that IP addresses are technically personally identifiable information. Yet again, this is something we're not tracking either. I'm assuming Userflow is then using some form of a cookie that does not track any personally identifiable information.

Here's a section relevant:

"What these two lines are stating is that cookies, insofar as they are used to identify users, qualify as personal data and are therefore subject to the GDPR. Companies do have a right to process their users’ data as long as they receive consent or if they have a legitimate interest."

We are not using cookies to identify users or qualify their personal data. To me, this means we are therefore _not_ subject to GDPR with this integration. As we are _also_ not receiving consent (because we're not collecting any data).

I think this concludes that we're in the clear.

File: .env.example [hunk 2]
```diff
@@ -40,5 +40,8 @@ SUBGRAPH_WS_ENDPOINT=ws://127.0.0.1:8001/subgraphs/name/joinColony/subgraph
 # Used in non-local network deployments to customize the network address
 # NETWORK_CONTRACT_ADDRESS=
 
-Used (currently) to set a manual provider for Xdai deployments
+# Used (currently) to set a manual provider for Xdai deployments
 # RPC_URL=
+
+# Token for "userflow.com" user onboarding
+USERFLOW_TOKEN=ct_snrheuld3nbe3exppc4foxcb54
```
Comment (rdig) [inline .env.example:47]: Please don't commit a "real" api key in the example environment variables

Just put in the variable's name:
```suggestion
USERFLOW_TOKEN=
```

**Comment by rdig** [comment]
Works for me.

@arrenv besides moving the init call to index, is there anything left to do here ?

File: .env.example (hunk 2 above)
Comment (arrenv) [reply .env.example:47]: Yeah, I think I just misunderstood your comment in the other related PR. Cleaning up now.

**Comment by arrenv** [comment]
@rdig We should be good to go now, I have updated based on all comments.

**Review by rdig — APPROVED** [review]
Goof to go
